### PR TITLE
Bump gc ruboconfig to v2.3.4

### PR DESCRIPTION
**Pull request by timrogers**

### CODE REVIEW

**Comment by timrogers** [comment]
@lawrencejones Ready for review!

File: spec/lib/coach/handler_spec.rb [hunk 1]
```diff
@@ -5,27 +5,30 @@
...
   let(:middleware_d) { build_middleware("D") }
 
   let(:terminal_middleware) { build_middleware("Terminal") }
-  subject(:handler) { Coach::Handler.new(terminal_middleware, handler: true) }
 
   before { Coach::Notifications.unsubscribe! }
 
   describe "#call" do
-    let(:a_spy) { spy("middleware a") }
-    let(:b_spy) { spy("middleware b") }
```
Comment (lawrencejones) [inline spec/lib/coach/handler_spec.rb]: Wow. This was some of the first rspec I'd ever written. I forgot spy's even exist!

File: spec/lib/coach/handler_spec.rb [hunk 2]
```diff
@@ -56,11 +61,12 @@
...
-      before { middleware_b.uses(middleware_c) }
-      before { terminal_middleware.uses(middleware_a) }
-      before { terminal_middleware.uses(middleware_b) }
-
+      before do
+        middleware_a.uses(middleware_c)
+        middleware_b.uses(middleware_c)
+        terminal_middleware.uses(middleware_a)
+        terminal_middleware.uses(middleware_b)
+      end
```
Comment (lawrencejones) [inline spec/lib/coach/handler_spec.rb]: Needs newline

File: spec/lib/coach/middleware_spec.rb [hunk 3]
```diff
@@ -17,16 +17,16 @@
       before { middleware_class.provides(:foo, :bar) }
 
       it "returns true" do
-        expect(middleware_class.provides?(:foo)).to be_truthy
-        expect(middleware_class.provides?(:bar)).to be_truthy
+        expect(middleware_class).to be_provides(:foo)
```
Comment (lawrencejones) [inline spec/lib/coach/middleware_spec.rb]: Urgh. This is weird. I wouldn't understand what this is doing by just looking at it.

Can we exclude this cop here? Or is it only me who struggles to parse this?

Comment (timrogers) [reply spec/lib/coach/middleware_spec.rb]: Didn't spot this. Definitely!

**Pull request merged**